File: src/main/java/org/dukcode/ps/codetree/trail02/chapter03/lesson01/IntroSortingWords.java
package org.dukcode.ps.codetree.trail02.chapter03.lesson01;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.Arrays;

/**
 * @see <a
 * href="https://www.codetree.ai/ko/trails/complete/curated-cards/intro-sorting-words/description">LINK</a>
 */
public class IntroSortingWords {

  private static BufferedReader br;
  private static BufferedWriter bw;

  public static void main(String[] args) throws IOException {
    br = new BufferedReader(new InputStreamReader(System.in));
    bw = new BufferedWriter(new OutputStreamWriter(System.out));

    int n = Integer.parseInt(br.readLine());
    String[] words = new String[n];
    for (int i = 0; i < n; i++) {
      words[i] = br.readLine();
    }

    Arrays.sort(words);

    for (String word : words) {
      bw.write(word);
      bw.newLine();
    }

    br.close();
    bw.close();

  }
}

File: src/main/kotlin/org/dukcode/ps/codetree/trail02/chapter03/lesson01/IntroSortingWords.kt
package org.dukcode.ps.codetree.trail02.chapter03.lesson01.introSortingWords

import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import java.util.StringTokenizer

private val br = BufferedReader(InputStreamReader(System.`in`))
private val bw = BufferedWriter(OutputStreamWriter(System.out))

/**
 * @see <a href="https://www.codetree.ai/ko/trails/complete/curated-cards/intro-sorting-words/description">LINK</a>
 */
fun main() {
    val n = br.readLine().toInt()
    val words =
        MutableList(n) { br.readLine() }.apply { sort() }

    for (word in words) {
        bw.write(word)
        bw.newLine()
    }

    bw.flush()
}

private fun parseInts() =
    StringTokenizer(br.readLine()).run {
        IntArray(countTokens()) {
            nextToken().toInt()
        }
    }
